bundles/util/tools.vitruv.applications.util.temporary/src/tools/vitruv/applications/util/temporary/java/JavaTypeUtil.xtend

```diff
@@ -0,0 +1,353 @@
+     */
+    def private static dispatch Type getTypeFromReference(TypeReference typeReference) {
+        logger.warn(typeReference + " is neither a NamespaceClassifierReference nor a PrimitiveType. Returning null.")
+        return null
+    }
+
+    def private static dispatch Type getTypeFromReference(Void nullTypeReference) {
+        logger.warn("Cannot get Type of a null-TypeReference. Returning null.")
+        return null
+    }
+
+    def private static dispatch Type getTypeFromReference(PrimitiveType primitiveType) {
+        return primitiveType
+    }
+
+    def private static dispatch Type getTypeFromReference(ClassifierReference classifierReference) {
+        return classifierReference.target
+    }
+
+    def private static dispatch Type getTypeFromReference(NamespaceClassifierReference namespace) {
+        if (namespace === null || namespace.classifierReferences.nullOrEmpty) {
+            return null
+        }
+        return getTypeFromReference(namespace.classifierReferences.head)
+    }
+
+    /**
+     * Retrieves the classifier wrapped in a type reference.
+     * @return the classifier or null if the type reference does not contain any classifier.
+     */
+    def static Classifier getClassifierFromTypeReference(TypeReference typeRef) {
+        val type = getTypeFromReference(typeRef)
+        if (type instanceof Classifier) {
+            return type
+        }
+        logger.warn("The TypeReference " + typeRef + " does not contain a Classifier. Returning null.")
+        return null
+    }
+
+    /**
+     * Retrieves the classifier wrapped in a type reference. Proxys are resolved and the classifier URI is normalized.
+     * @return the classifier or null if the type reference does not contain any classifier.
+     */
+    def static Classifier getNormalizedClassifierFromTypeReference(TypeReference typeRef) {
+        var type = getTypeFromReference(typeRef)
+        if (type instanceof Classifier) {
+            if (type.eIsProxy) { // resolve proxy
+                val resourceSet = type.eResource?.resourceSet // resource can be null, but EcoreUtil.resolve() can handle null as resource set
+                type = EcoreUtil.resolve(type, resourceSet) as Classifier
+            }
+            normalizeURI(type)
+            return type as Classifier
+        }
+        logger.warn("The TypeReference " + typeRef + " does not contain a Classifier. Returning null.")
+        return null
+    }
+
+    /**
+     * Retrieves the interface wrapped in a type reference.
+     * @return the interface or null if the type reference does not contain any interfaces.
+     */
+    def static Interface getInterfaceFromTypeReference(TypeReference typeRef) {
+        val type = getTypeFromReference(typeRef)
+        if (type instanceof Interface) {
+            return type
+        }
+        logger.warn("The TypeReference " + typeRef + " does not contain a Interface. Returning null.")
+        return null
+    }
```

Comment:
These methods were the ones causing the issues we discussed. This functionality was available in four different utility classes with slightly different implementations. Now we have these three utility methods for getting classifiers from type references. The underlying dispatch methods `getTypeFromReference` are now private as they no longer need to be exposed.